core/include/detray/definitions/detail/accessor.hpp

```
@@ -41,6 +42,11 @@ namespace detail {
 using std::get;
 using thrust::get;
 
+template <std::size_t id>
```

Why does this need a template?

---

Because ```get``` is always called with a template argument, so this has to be added as a dummy non-type here, too

---

OK. but still not sure if we need this function. Maybe you used different template parameter for a link type and need this one.

---

Correct. Sometimes in testing code we use dummy links (these are the same default mask ```volume_links``` I have to revise anyway), which get automatically resolved by this